W/WaveFD/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "WaveFD"
+version = v"0.1.0"
+
+# Collection of sources required to build AzStorage
+sources = [
+    GitSource(
+        "https://github.com/ChevronETC/WaveFD.jl.git",
+        "52105fc62f8a52ade84365f403a352a47e3909d6"
+    )
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/WaveFD.jl/src
+
+CXXFLAGS="-funroll-loops"
+if [[ "${target}" == i686-linux-gnu || "${target}" == x86_64-linux-gnu || "${target}" == powerpc6rle-linux-gnu ]]; then
+    CXXFLAGS+=" -D__FUNCTION_CLONES__"
+fi
+
+echo "target=$target, CXXFLAGS=$CXXFLAGS"
+
+cmake . -DCMAKE_CXX_FLAGS="${CXXFLAGS}" -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake -DCMAKE_BUILD_TYPE=Release
+make VERBOSE=1
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libprop2DAcoIsoDenQ_DEO2_FDTD", :libprop2DAcoIsoDenQ_DEO2_FDTD),
+    LibraryProduct("libprop2DAcoVTIDenQ_DEO2_FDTD", :libprop2DAcoVTIDenQ_DEO2_FDTD),
+    LibraryProduct("libprop2DAcoTTIDenQ_DEO2_FDTD", :libprop2DAcoTTIDenQ_DEO2_FDTD),
+    LibraryProduct("libprop3DAcoIsoDenQ_DEO2_FDTD", :libprop3DAcoIsoDenQ_DEO2_FDTD),
+    LibraryProduct("libprop3DAcoVTIDenQ_DEO2_FDTD", :libprop3DAcoVTIDenQ_DEO2_FDTD),
+    LibraryProduct("libprop3DAcoTTIDenQ_DEO2_FDTD", :libprop3DAcoTTIDenQ_DEO2_FDTD),
+    LibraryProduct("libillumination", :libillumination),
+    LibraryProduct("libspacetime", :libspacetime)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("CompilerSupportLibraries_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9")
```

Comment:
Any particular reason tu use GCC v9?  This won't work with any version of Julia

Reply:
not really.  It's just what we have being testing with.  What is maximum version that works with any version of Julia?  What is the minimum version of Julia that works with gcc 9?

Reply:
Since this is a C++ library, it really depends on the version of libstdc++ which comes with Julia and there isn't a single good answer.  For example, the official binaries use GCC v7 on Linux, but those who build Julia from source will get whatever libstdc++ is on their system.

If you want maximum compatibility, you need to use the oldest GCC version possible, that is just don't specify it at all.

Reply:
Actually, it might even work with GCC v9:
```
% nm *.so|c++filt|grep GLIBCXX 
                 U operator delete[](void*)@@GLIBCXX_3.4
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new[](unsigned long)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
                 U operator delete[](void*)@@GLIBCXX_3.4
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new[](unsigned long)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
                 U operator delete[](void*)@@GLIBCXX_3.4
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new[](unsigned long)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
                 U operator delete[](void*)@@GLIBCXX_3.4
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new[](unsigned long)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
                 U operator delete[](void*)@@GLIBCXX_3.4
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new[](unsigned long)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
                 U operator delete[](void*)@@GLIBCXX_3.4
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new[](unsigned long)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
```
There doesn't seem to be any reference to the patch version of GLIBCXX :thinking:

Ok, I'll merge, but if you get reports of users that can't run it because of undefined references to GLIBCXX symbols, you know what's wrong

Reply:
awesome.  thanks.  I notice that the FFTW build script uses GCC 8, so I'll try to spend some time testing with version 8.

Reply:
I think FFTW doesn't use C++ though, the compatibility problem is with libstdc++